java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -51,7 +47,7 @@
 import java.util.List;
 
 @Rule(key = "S3518")
-public class DivisionByZeroCheck extends SECheck {
+public class DivisionByZeroCheck extends ExceptionalEndOfExecutionCheck {
```

Comment:
please favor composition over inheritance : that would simplify the code and avoid a complex uneeded hierarchy